test/Common/TaskExtensions.cs

```diff
@@ -18,7 +20,7 @@ public static Task OrTimeout(this Task task, int milliseconds = DefaultTimeout,
 
         public static async Task OrTimeout(this Task task, TimeSpan timeout, [CallerMemberName] string memberName = null, [CallerFilePath] string filePath = null, [CallerLineNumber] int? lineNumber = null)
         {
-            var completed = await Task.WhenAny(task, Task.Delay(timeout));
+            var completed = await Task.WhenAny(task, Task.Delay(Debugger.IsAttached ? Timeout.InfiniteTimeSpan : timeout));
```

Comment:
Yea, I gave up. We're keeping it. Our product is impossible to debug without it.